lasagna-master/hemartin/lasagna_master.go

```diff
@@ -0,0 +1,70 @@
+package lasagna
+
+const noodleQuantityPerLayer int = 50
+const sauceQuantityPerLayer float64 = 0.2
+const originalQuantityPortions float64 = 2.0
+
+func PreparationTime(layers []string, layerPrepTime int) int {
+	if layerPrepTime <= 0 {
+		layerPrepTime = 2
+	}
+	return layerPrepTime * len(layers)
+
+}
+
+func Quantities(layers []string) (int, float64) {
+	return QuantitiesFast(layers)
+}
+
+// This was the first implementation
+func QuantitiesSlow(layers []string) (int, float64) {
+	count := map[string]int{
+		"noodles": 0,
+		"sauce":   0,
+	}
+
+	for _, layer := range layers {
+		if layer == "noodles" || layer == "sauce" {
+			count[layer]++
+		}
+
+	}
+
+	return count["noodles"] * noodleQuantityPerLayer, float64(count["sauce"]) * sauceQuantityPerLayer
+}
+
+// Performant implementation
+func QuantitiesFast(layers []string) (int, float64) {
+	noodles := 0
+	sauce := 0.0
+
+	for _, layer := range layers {
+		if layer == "noodles" {
+			noodles += noodleQuantityPerLayer
+			continue
+		}
+
+		if layer == "sauce" {
+			sauce += sauceQuantityPerLayer
+			continue
+		}
```

Comment:
This would be better with a switch block.

Reply:
Indeed, I changed it.